viam-cartographer/cartographer_patches/carto.patch

```diff
@@ -0,0 +1,304 @@
+-  list(REMOVE_ITEM TEST_LIBRARY_HDRS ${ALL_PROMETHEUS_FILES})
+-  list(REMOVE_ITEM TEST_LIBRARY_SRCS ${ALL_PROMETHEUS_FILES})
+-  list(REMOVE_ITEM ALL_TESTS ${ALL_PROMETHEUS_FILES})
+-  list(REMOVE_ITEM ALL_EXECUTABLES ${ALL_PROMETHEUS_FILES})
+-endif()
++
++# GRPC files are part of the cartographer library - but we don't need them.
++list(REMOVE_ITEM ALL_LIBRARY_HDRS ${ALL_GRPC_FILES})
++list(REMOVE_ITEM ALL_LIBRARY_SRCS ${ALL_GRPC_FILES})
++list(REMOVE_ITEM TEST_LIBRARY_HDRS ${ALL_GRPC_FILES})
++list(REMOVE_ITEM TEST_LIBRARY_SRCS ${ALL_GRPC_FILES})
++list(REMOVE_ITEM ALL_TESTS ${ALL_GRPC_FILES})
++list(REMOVE_ITEM ALL_EXECUTABLES ${ALL_GRPC_FILES})
++
++# Prometheus files are part of the cartographer library - but we don't need them.
++list(REMOVE_ITEM ALL_LIBRARY_HDRS ${ALL_PROMETHEUS_FILES})
++list(REMOVE_ITEM ALL_LIBRARY_SRCS ${ALL_PROMETHEUS_FILES})
++list(REMOVE_ITEM TEST_LIBRARY_HDRS ${ALL_PROMETHEUS_FILES})
++list(REMOVE_ITEM TEST_LIBRARY_SRCS ${ALL_PROMETHEUS_FILES})
++list(REMOVE_ITEM ALL_TESTS ${ALL_PROMETHEUS_FILES})
++list(REMOVE_ITEM ALL_EXECUTABLES ${ALL_PROMETHEUS_FILES})
++
+ set(INSTALL_SOURCE_HDRS ${ALL_LIBRARY_HDRS} ${TEST_LIBRARY_HDRS})
+-file(GLOB_RECURSE INTERNAL_HDRS "cartographer/*/internal/*.h")
+-list(REMOVE_ITEM INSTALL_SOURCE_HDRS ${INTERNAL_HDRS})
++# file(GLOB_RECURSE INTERNAL_HDRS "cartographer/*/internal/*.h")
++# list(REMOVE_ITEM INSTALL_SOURCE_HDRS ${INTERNAL_HDRS})
+ 
+ file(GLOB_RECURSE ALL_PROTOS "cartographer/*.proto")
+ file(GLOB_RECURSE ALL_GRPC_SERVICES "cartographer/*_service.proto")
+@@ -155,33 +118,6 @@ set_source_files_properties(${ALL_PROTO_SRCS} ${ALL_PROTO_HDRS} PROPERTIES GENER
+ list(APPEND ALL_LIBRARY_HDRS ${ALL_PROTO_HDRS})
+ list(APPEND ALL_LIBRARY_SRCS ${ALL_PROTO_SRCS})
+ 
+-if(${BUILD_GRPC})
+-  set(ALL_GRPC_SERVICE_SRCS)
+-  set(ALL_GRPC_SERVICE_HDRS)
+-  foreach(ABS_FIL ${ALL_GRPC_SERVICES})
+-    file(RELATIVE_PATH REL_FIL ${PROJECT_SOURCE_DIR} ${ABS_FIL})
+-    get_filename_component(DIR ${REL_FIL} DIRECTORY)
+-    get_filename_component(FIL_WE ${REL_FIL} NAME_WE)
+-
+-    list(APPEND ALL_GRPC_SERVICE_SRCS "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.cc")
+-    list(APPEND ALL_GRPC_SERVICE_HDRS "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.h")
+-
+-    add_custom_command(
+-      OUTPUT "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.cc"
+-             "${PROJECT_BINARY_DIR}/${DIR}/${FIL_WE}.pb.h"
+-      COMMAND  ${PROTOBUF_PROTOC_EXECUTABLE}
+-      ARGS --cpp_out  ${PROJECT_BINARY_DIR}
+-        -I ${PROJECT_SOURCE_DIR}
+-        ${ABS_FIL}
+-      DEPENDS ${ABS_FIL}
+-      COMMENT "Running C++ protocol buffer compiler on ${ABS_FIL}"
+-      VERBATIM
+-    )
+-  endforeach()
+-  set_source_files_properties(${ALL_GRPC_SERVICE_SRCS} ${ALL_GRPC_SERVICE_HDRS} PROPERTIES GENERATED TRUE)
+-  list(APPEND ALL_LIBRARY_HDRS ${ALL_GRPC_SERVICE_HDRS})
+-  list(APPEND ALL_LIBRARY_SRCS ${ALL_GRPC_SERVICE_SRCS})
+-endif()
+ set(INSTALL_GENERATED_HDRS ${ALL_PROTO_HDRS} ${ALL_GRPC_SERVICE_HDRS})
```

Comment:
[q] is this needed after all other references to ALL_GRPC_SERVICE_HDRS have been deleted?

Reply:
Per previous, not my changes, so can't say for sure. I suspect it's not though. 

Cleaning this up could be good for a followup task on your team. You can checkout the original version just add a more minimal or cleaner set of changes.